elide-core/src/test/java/com/yahoo/elide/core/UpdateOnCreateTest.java

```diff
@@ -0,0 +1,242 @@
+
+        when(tx.createNewObject(UpdateAndCreate.class)).thenReturn(updateAndCreateNewObject);
+        when(tx.loadObject(eq(UpdateAndCreate.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(updateAndCreateExistingObject);
+        when(tx.loadObject(eq(Book.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(book);
+        when(tx.loadObject(eq(Author.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(author);
+    }
+
+    @Test
+    public void testUpdateOnCreateUserOne() {
+
+        PersistentResource<UpdateAndCreate> created = PersistentResource.createObject(null, UpdateAndCreate.class, userOneScope, Optional.of("uuid"));
+        PersistentResource<Book> loadedBook = PersistentResource.loadRecord(Book.class,
+                "1",
+                userOneScope);
+        PersistentResource<Author> loadedAuthor = PersistentResource.loadRecord(Author.class,
+                "1",
+                userOneScope);
+
+        created.updateAttribute("name", "");
+        created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        Assert.assertThrows(ForbiddenAccessException.class, () -> {
+            created.updateAttribute("type", UpdateAndCreate.AuthorType.CONTRACTED);
+            created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        });
+        created.updateAttribute("alias", "");
+        created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        created.addRelation("books", loadedBook);
+        created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        Assert.assertThrows(ForbiddenAccessException.class, () -> {
+            created.addRelation("author", loadedAuthor);
+            created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+
+        });
+    }
+
+    @Test
+    public void testNormalUpdateUserOne() {
+        PersistentResource<UpdateAndCreate> loaded = PersistentResource.loadRecord(UpdateAndCreate.class,
+                "1",
+                userOneScope);
+        PersistentResource<Book> loadedBook = PersistentResource.loadRecord(Book.class,
+                "1",
+                userOneScope);
+        PersistentResource<Author> loadedAuthor = PersistentResource.loadRecord(Author.class,
+                "1",
+                userOneScope);
+
+        Assert.assertThrows(ForbiddenAccessException.class, () -> {
+            loaded.updateAttribute("name", "");
+            loaded.getRequestScope().getPermissionExecutor().executeCommitChecks();
```

Comment:
Both `updateAttribute()` and `executeCommitChecks()` throws `ForbiddenAccessException`, so I guess you should make two assertions to check which one is actually throwing. Same to all following.